### PR TITLE
This patch updates the readme, the run_llama_train.sh, and the train.py files

### PR DESCRIPTION
**Pull request by sarabiap**
Tokenizer changes impacting readme and run_llama_train.sh
- We already have the tokenizer downloaded in /app/torchtitan/datasets/tokenizer/ so we can remove the comments and the python command from the shell script — python torchtitan/datasets/download_tokenizer.py --repo_id meta-llama/Meta-Llama-3.1-8B --tokenizer_path "original" --local_dir=/app/torchtitan/datasets/tokenizer/ --hf_token=$HF_TOKEN
- This also removes the requirement to export the $HF_Token environment variable in the readme — I've moved the entire Export HF token section to a deprecated section at the end of the readme and subsequently removed the env vars in the CLI commands
- I also updated the download llama tokenizer section to note that it's already in the repo

Adding Nsight Systems instructions
-  This is a new section in the readme

Cleaned up the torchrun command
- This is an aesthetics clean up

Included clip gradients in train.py script
- Added this in line 323

Readme portforward instructions
- Added instructions here using the runai CLI and kubectl

NGC API Key and CLI instructions
- Added instructions to resolve issues encountered by the customer e.g. 403 error

### CODE REVIEW

File: train.py [hunk 1]
```diff
@@ -320,12 +320,20 @@ def main(job_config: JobConfig):
                     del pred
                     loss.backward()
 
-            # optimizer step
+            # Clip Gradients
+            utils.clip_grad_norm_(
```
Comment (smcdonald248) [inline train.py]: This block was commented out -- for some specific reason?

**Comment by smcdonald248** [comment]
Outside of the updates to train.py, i don't have any concerns. I didn't run through your nsys instructions yet, but will soon.
